src/tdyio.c

```diff
@@ -73,33 +74,48 @@ PetscErrorCode TDyIOSetMode(TDy tdy, TDyIOFormat format){
 PetscErrorCode TDyIOWriteVec(TDy tdy){
   PetscErrorCode ierr;
   PetscBool useNatural;
-  
-  Vec v = tdy->solution;
+  Vec s;
+  PetscReal *s_vec_ptr;
+  PetscInt cStart,cEnd,c,i;
+  Vec p = tdy->solution;
   DM dm = tdy->dm;
   PetscReal time = tdy->ti->time;
- 
+
+  ierr = DMCreateGlobalVector(dm, &s);
+  ierr = VecGetArray(s,&s_vec_ptr);
+  ierr = DMPlexGetHeightStratum(tdy->dm,0,&cStart,&cEnd); CHKERRQ(ierr);
+  for (c=cStart;c<cEnd;c++) {
+    i = c-cStart;
+    s_vec_ptr[i] = tdy->cc->S[i];
+  }
+  ierr = VecRestoreArray(s,&s_vec_ptr);CHKERRQ(ierr);
+
   if (tdy->io->format == PetscViewerASCIIFormat) {
-    ierr = TDyIOWriteAsciiViewer(v, time);CHKERRQ(ierr);
+    ierr = TDyIOWriteAsciiViewer(p,s,time);CHKERRQ(ierr);
   }
   else if (tdy->io->format == ExodusFormat) {
     char *ofilename = tdy->io->filename;
 
     ierr = TdyIOAddExodusTime(ofilename,time,tdy->io);CHKERRQ(ierr);
-    ierr = TdyIOWriteExodusVar(ofilename,dm,v,tdy->io,time);CHKERRQ(ierr);
+    ierr = TdyIOWriteExodusVar(ofilename,dm,p,s,tdy->io,time);CHKERRQ(ierr);
```

Comment:
Instead of increasing the input argument (`s` for saturation), have two separate calls to the function to write `p` and `s` e.g.


```

ierr = TDyIOWriteExodusVar(filename, dm, P, zonalVarNames[0], tdy->io,time);CHKERRQ(ierr);
ierr = TDyIOWriteExodusVar(filename, dm, S, zonalVarNames[1], tdy->io,time);CHKERRQ(ierr);
...
...

// Update function
PetscErrorCode TDyIOWriteExodusVar(char *ofilename, DM dm, 
  Vec var, char *varname, PetscInt sequence_num, PetscReal time){
...
...
}
```